### PR TITLE
PUBDEV-6280: Documentation for negative binomial family in GLM

### PR DESCRIPTION
**Pull request by angela0xdata**
- Updated GLM documentation to include new negative binomial family
- Updated "links" table to include the negative binomial family
- Added new "theta" parameter to GLM and Flow chapters. As this is gridable, also added this to the Grid Search topic.
Driveby fixes: Added internal links to different family topics.

TODO: Update booklet after review

### CODE REVIEW

**Comment by angela0xdata** [comment]
Here's what the new section looks like.
Screen Shot 2019-03-20 at 1 52 53 PM

**Comment by wendycwong** [comment]
Angela

It is totally my fault, I have some typos in the documentation.  I am going to slack you the new ones.  Sorry.

**Comment by angela0xdata** [comment]
Updated some equations

Screen Shot 2019-03-20 at 3 00 24 PM

**Comment by wendycwong** [comment]
Angela:

There is a problem with this equation:
image

The ui are not a subscript to the theta.  This is the correct look:
image

**Comment by wendycwong** [comment]
Thank you so much Angela.

**Comment by angela0xdata** [comment]
Is this correct?

Screen Shot 2019-03-20 at 3 23 38 PM

File: h2o-docs/src/product/data-science/glm.rst [hunk 1]
```diff
@@ -198,14 +203,15 @@ Families
...
-- ``tweedie``: The data must be numeric and continuous (Real) and non-negative.
+- ``gaussian``: (See `Linear Regression (Gaussian Family)`_.) The response must be numeric (Real or Int). This is the default family.
+- ``binomial``: (See `Logistic Regression (Binomial Family)`_). The response must be categorical 2 levels/classes or binary (Enum or Int).
+- ``ordinal``: (See `Logistic Ordinal Regression (Ordinal Family)`_). Requires a categorical response with at least 3 levels. (For 2-class problems, use family="binomial".)
+- ``quasibinomial``: (See `Pseudo-Logistic Regression (Quasibinomial Family)`_). The response must be numeric.
+- ``multinomial``: (See `Multiclass Classification (Multinomial Family)`_). The response can be categorical with more than two levels/classes (Enum).
+- ``poisson``: (See `Poisson Models`_). The response must be numeric and non-negative (Int).
+- ``gamma``: (See `Gamma Models`_). The response must be numeric and continuous and positive (Real or Int).
+- ``tweedie``: (See `Tweedie Models`_). The response must be numeric and continuous (Real) and non-negative.
+- ``negativebinomial``: (See `Negative Binomial Models`_). The response must be numeric.
```
Comment (wendycwong) [inline h2o-docs/src/product/data-science/glm.rst]: Same here, please say 

The response must be numeric and non-negative (Int).

File: h2o-docs/src/product/data-science/glm.rst [hunk 2]
```diff
@@ -66,21 +67,24 @@ Defining a GLM Model
...
-   -  If the family is **poisson**, the data must be numeric and non-negative (**Int**).
-   -  If the family is **gamma**, the data must be numeric and continuous and positive (**Real** or **Int**).
-   -  If the family is **tweedie**, the data must be numeric and continuous (**Real**) and non-negative.
+   -  If the family is **gaussian**, the response must be numeric (**Real** or **Int**). (default)
+   -  If the family is **binomial**, the response must be categorical 2 levels/classes or binary (**Enum** or **Int**).
+   -  If the family is **multinomial**, the response can be categorical with more than two levels/classes (**Enum**).
+   -  If the family is **ordinal**, the response must be categorical with at least 3 levels.
+   -  If the family is **quasibinomial**, the response must be numeric.
+   -  If the family is **poisson**, the response must be numeric and non-negative (**Int**).
+   -  If the family is **negativebinomial**, the response must be numeric (**Int**).
```
Comment (wendycwong) [inline h2o-docs/src/product/data-science/glm.rst]: Please change this:  the response must be numeric (**Int**).
to: the response must be numeric and non-negative (**Int**).

Comment (angela0xdata) [reply h2o-docs/src/product/data-science/glm.rst]: Fixed!

**Pull request merged**